common/include/pcl/common/impl/io.hpp

```diff
@@ -68,10 +63,11 @@ pcl::getFieldIndex (const std::string &field_name,
   fields.clear ();
   // Get the fields list
   pcl::for_each_type<typename pcl::traits::fieldList<PointT>::type>(pcl::detail::FieldAdder<PointT>(fields));
-  for (std::size_t d = 0; d < fields.size (); ++d)
-    if (fields[d].name == field_name)
-      return (static_cast<int>(d));
-  return (-1);
+  const auto result = std::find_if(fields.begin (), fields.end (),
+      [&field_name](const auto field) { return field.name == field_name; });
```

Comment:
Wouldn't `const auto&` be better here?

Reply:
Please do keep an eye out for such issues